haystack/preview/components/generators/openai/chatgpt.py

```diff
@@ -0,0 +1,194 @@
+from typing import Optional, List, Callable, Dict, Any
+
+import logging
+
+from haystack.preview import component, default_from_dict, default_to_dict
+from haystack.preview.llm_backends.openai.chatgpt import ChatGPTBackend
+from haystack.preview.llm_backends.chat_message import ChatMessage
+from haystack.preview.llm_backends.openai._helpers import default_streaming_callback
+
+
+logger = logging.getLogger(__name__)
+
+
+TOKENS_PER_MESSAGE_OVERHEAD = 4
+
+
+@component
+class ChatGPTGenerator:
+    """
+    ChatGPT LLM Generator.
+
+    Queries ChatGPT using OpenAI's GPT-3 ChatGPT API. Invocations are made using REST API.
+    See [OpenAI ChatGPT API](https://platform.openai.com/docs/guides/chat) for more details.
+    """
+
+    # TODO support function calling!
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        model_name: str = "gpt-3.5-turbo",
+        system_prompt: Optional[str] = None,
+        max_tokens: Optional[int] = 500,
+        temperature: Optional[float] = 0.7,
+        top_p: Optional[float] = 1,
+        n: Optional[int] = 1,
+        stop: Optional[List[str]] = None,
+        presence_penalty: Optional[float] = 0,
+        frequency_penalty: Optional[float] = 0,
+        logit_bias: Optional[Dict[str, float]] = None,
+        stream: bool = False,
+        streaming_callback: Optional[Callable] = default_streaming_callback,
+        api_base_url: str = "https://api.openai.com/v1",
+        openai_organization: Optional[str] = None,
+    ):
+        """
+        Creates an instance of ChatGPTGenerator for OpenAI's GPT-3.5 model.
+
+        :param api_key: The OpenAI API key.
+        :param model_name: The name or path of the underlying model.
+        :param system_prompt: The prompt to be prepended to the user prompt.
+        :param max_tokens: The maximum number of tokens the output text can have.
+        :param temperature: What sampling temperature to use. Higher values means the model will take more risks.
+            Try 0.9 for more creative applications, and 0 (argmax sampling) for ones with a well-defined answer.
+        :param top_p: An alternative to sampling with temperature, called nucleus sampling, where the model
+            considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens
+            comprising the top 10% probability mass are considered.
+        :param n: How many completions to generate for each prompt.
+        :param stop: One or more sequences where the API will stop generating further tokens.
```

Comment:
Does this define some specific phrase that tells the API to stop generating tokens?

Reply:
I changed the wording to try make it more clear, it was quite ambiguous indeed